src/ctrlc.rs

```diff
@@ -0,0 +1,86 @@
+use console::Term;
+use once_cell::sync::Lazy;
+use signal_hook::{
+    consts::SIGINT,
+    iterator::{Handle, Signals},
+};
+use std::{
+    io::Error,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Mutex, RwLock,
+    },
+    thread,
+    time::Duration,
+};
+
+static MUTEX: Mutex<()> = Mutex::new(());
+static INIT: AtomicBool = AtomicBool::new(false);
+static HANDLE: Lazy<RwLock<Option<CtrlcHandle>>> = Lazy::new(|| RwLock::new(None));
+
+#[derive(Clone)]
+pub struct CtrlcHandle(Handle);
+
+impl CtrlcHandle {
+    #[allow(dead_code)]
+    pub fn close(&self) {
+        self.0.close();
+    }
+}
+
+impl Drop for CtrlcHandle {
+    fn drop(&mut self) {
+        // let the signal handler finish
+        thread::sleep(Duration::from_millis(100));
```

Comment:
`drop` works too well and executes even before the closure is called or done executing

Reply:
hmm might be better to go back to the way you had it before because this might not complete before the CLI stops running